core/jvm/src/test/scala/zio/IOSpec.scala

```diff
@@ -579,4 +583,39 @@ class IOSpec(implicit ee: org.specs2.concurrent.ExecutionEnv) extends TestRuntim
 
     goodCase and partialBadCase and badCase
   }
+
+  def testForeachParN_Threads = {
```

Comment:
How long do these tests take? If it's close to the timeout, the test will be flaky on CI.

Reply:
The one with a collection size of 100,000 takes about 2 seconds on my local machine. The challenge here is that you need quite a large collection size to verify the improvement from the previous implementation. On my local machine the previous implementation would work on collection sizes up to 50,000. Not sure how to test this more efficiently.